recipes/systemc/2.3.3/conanfile.py

```diff
@@ -0,0 +1,90 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class SystemcConan(ConanFile):
+    name = "systemc"
+    version = "2.3.3"
+    description = """SystemC is a set of C++ classes and macros which provide
+                     an event-driven simulation interface."""
+    homepage = "https://www.accellera.org/"
+    url = "https://github.com/conan-io/conan-center-index "
+    license = "MIT"
+    topics = ("simulation", "modeling", "esl", "tlm")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "disable_async_updates": [True, False],
+        "disable_copyright_msg": [True, False],
+        "disable_virtual_bind": [True, False],
+        "enable_assertions": [True, False],
+        "enable_immediate_self_notifications": [True, False],
+        "enable_pthreads": [True, False]
+    }
+    default_options = {
+        "shared": True,
+        "disable_async_updates": False,
+        "disable_copyright_msg": False,
+        "disable_virtual_bind":  False,
+        "enable_assertions": True,
+        "enable_immediate_self_notifications": False,
+        "enable_pthreads": False
+    }
+    generators = "cmake"
+    exports_sources = "sc_string_view.patch"
+
+    def configure(self):
+
+        if tools.valid_min_cppstd(self, "17"):
+            raise ConanInvalidConfiguration(
+                "C++ Standard %s not supported by SystemC" %
+                self.settings.compiler.cppstd)
+
+        if not self.settings.compiler.cppstd:
+            if tools.valid_min_cppstd(self, "14"):
+                self.settings.compiler.cppstd = 14
+            elif tools.valid_min_cppstd(self, "11"):
+                self.settings.compiler.cppstd = 11
+            elif tools.valid_min_cppstd(self, "98"):
+                self.settings.compiler.cppstd = 98
```

Comment:
If I remember correct, systemc is very pedantic with the c++ standard. It encodes the c++ standard in its namespace/symbols.
I would just ignore `self.settings.compiler.cppstd` in the `configure()` method and do the following in `build()`:
```python
if self.settings.compiler.cppstd:
    cmake.definitions["CMAKE_CXX_STANDARD"] = self.settings.compiler.cppstd
```
If `self.settings.compiler.cppstd` is `None`, then the default c++ standard of that compiler version is used, which should always be the same for a certain compiler version.

Reply:
Yes it is really picky about the version, I will give this a go.
I assume the max version check at line 38 is OK?